UPGRADE.md

```diff
@@ -117,20 +117,18 @@ const messages = {
 const App = () => <Admin locale="en" messages={messages} />;
 
 // after
-import { Admin, GET_DEFAULT_MESSAGES, GET_LOCALE_MESSAGES } from 'react-admin';
+import { Admin } from 'react-admin';
 import enMessages from 'ra-language-english';
 import frMessages from 'ra-language-french';
 
-const messages = {
-    en: enMessages,
-    fr: frMessages,
-};
-
-const i18nProvider = (type, params) => {
-    if (type === GET_DEFAULT_MESSAGES) {
-        return messages.en; // The default messages should always resolve synchronously!! 
+const i18nProvider = locale => {
+    if (locale === 'fr') {
+        return frMessages;
     }
-    return messages[params.locale];
+
+    // If locale wasn't specified it's a request for default messages
+    // If locale is not supported, fallback to english
+    return enMessages;
```

Comment:
why don't you simply return `messages[locale]` all the time?

Reply:
For the fallback ?